src/custom/state/lists/updater/updaterMod.ts

```diff
@@ -0,0 +1,96 @@
+  // MOD: chainId
+  // const { library } = useActiveWeb3React()
+  const { chainId = DEFAULT_NETWORK_FOR_LISTS, library } = useActiveWeb3React()
+  const dispatch = useDispatch<AppDispatch>()
+  const isWindowVisible = useIsWindowVisible()
+
+  // get all loaded lists, and the active urls
+  const lists = useAllLists()
+  const activeListUrls = useActiveListUrls()
+
+  // initiate loading
+  useAllInactiveTokens()
+
+  const fetchList = useFetchListCallback()
+  const fetchAllListsCallback = useCallback(() => {
+    if (!isWindowVisible) return
+    Object.keys(lists).forEach(url =>
+      fetchList(url).catch(error => console.debug('interval list fetching error', error))
+    )
+  }, [fetchList, isWindowVisible, lists])
+
+  // fetch all lists every 10 minutes, but only after we initialize library
+  useInterval(fetchAllListsCallback, library ? 1000 * 60 * 10 : null)
+
+  // whenever a list is not loaded and not loading, try again to load it
+  useEffect(() => {
+    Object.keys(lists).forEach(listUrl => {
+      const list = lists[listUrl]
+      if (!list.current && !list.loadingRequestId && !list.error) {
+        fetchList(listUrl).catch(error => console.debug('list added fetching error', error))
+      }
+    })
+  }, [dispatch, fetchList, library, lists])
+
+  // automatically update lists if versions are minor/patch
+  useEffect(() => {
+    Object.keys(lists).forEach(listUrl => {
+      const list = lists[listUrl]
+      if (list.current && list.pendingUpdate) {
+        const bump = getVersionUpgrade(list.current.version, list.pendingUpdate.version)
+        switch (bump) {
+          case VersionUpgrade.NONE:
+            throw new Error('unexpected no version bump')
+          case VersionUpgrade.PATCH:
+          case VersionUpgrade.MINOR:
+            const min = minVersionBump(list.current.tokens, list.pendingUpdate.tokens)
+            // automatically update minor/patch as long as bump matches the min update
+            if (bump >= min) {
+              // MOD: need to update the acceptListUpdate params
+              // dispatch(acceptListUpdate(listUrl))
+              dispatch(acceptListUpdate({ chainId, url: listUrl }))
+            } else {
+              console.error(
+                `List at url ${listUrl} could not automatically update because the version bump was only PATCH/MINOR while the update had breaking changes and should have been MAJOR`
+              )
+            }
+            break
+
+          // update any active or inactive lists
+          case VersionUpgrade.MAJOR:
+            // MOD: need to update the acceptListUpdate params
+            // dispatch(acceptListUpdate(listUrl))
+            dispatch(acceptListUpdate({ chainId, url: listUrl }))
+        }
+      }
+    })
+    // MOD: deps
+    // }, [dispatch, lists, activeListUrls])
+  }, [dispatch, lists, activeListUrls, chainId])
+
+  // automatically initialise lists if chainId changes
+  useEffect(() => {
```

Comment:
Only update here that isn't adapting to new state shape

this calls list initialisation logic on chainId change